packages/cfd/src/Containers/cfd-password-modal/cfd-password-form/cfd-password-form.tsx

```diff
@@ -0,0 +1,288 @@
+    jurisdiction_selected_shortcode: string;
+    onCancel: () => void;
+    onForgotPassword: () => void;
+    should_set_trading_password: boolean;
+    show_eu_related_content: boolean;
+    submitPassword: TOnSubmitPassword;
+};
+
+const CFDPasswordForm = observer(
+    ({
+        account_title,
+        account_type,
+        closeModal,
+        error_message,
+        error_type,
+        form_error,
+        has_mt5_account,
+        is_real_financial_stp,
+        jurisdiction_selected_shortcode,
+        onCancel,
+        onForgotPassword,
+        platform,
+        should_set_trading_password,
+        show_eu_related_content,
+        submitPassword,
+        validatePassword,
+    }: TCFDPasswordFormProps) => {
+        const { ui } = useStore();
+        const { is_mobile } = ui;
+
+        const button_label = React.useMemo(() => {
+            if (error_type === PASSWORD_ERRORS.RESET) {
+                return localize('Try later');
+            }
+            return localize('Add account');
+        }, [error_type]);
+
+        const getCancelButtonLabel = ({
+            should_set_trading_password,
+            error_type,
+        }: Pick<TCFDPasswordFormProps, 'should_set_trading_password' | 'error_type'>) => {
+            if (should_set_trading_password && error_type !== PASSWORD_ERRORS.RESET) {
+                return is_mobile ? localize('Cancel') : null;
+            }
+
+            return localize('Forgot password?');
+        };
+
+        const has_cancel_button =
+            !is_mobile && (!should_set_trading_password || error_type !== PASSWORD_ERRORS.PASSWORD);
+
+        const cancel_button_label = getCancelButtonLabel({ should_set_trading_password, error_type });
+
+        const handleCancel = () => {
+            if (!has_cancel_button) {
+                return undefined;
+            }
+            if (should_set_trading_password) {
+                return onCancel();
+            }
+
+            return onForgotPassword();
+        };
+
+        const getAccountTitle = () => {
+            if (platform === CFD_PLATFORMS.DXTRADE) {
+                return getDxCompanies()[account_type.category as keyof TDxCompanies][
+                    account_type.type as keyof TDxCompanies['demo' | 'real']
+                ].short_title;
+            }
+
+            return account_title;
+        };
+
+        const handlePasswordInputChange = (
+            e: React.ChangeEvent<HTMLInputElement>,
+            handleChange: (el: React.ChangeEvent<HTMLInputElement>) => void,
+            validateForm: (values?: TCFDPasswordFormValues) => Promise<FormikErrors<TCFDPasswordFormValues>>,
+            setFieldTouched: (field: string, isTouched?: boolean, shouldValidate?: boolean) => void
+        ) => {
+            handleChange(e);
+            validateForm().then(() => {
+                setFieldTouched('password', true);
+            });
+        };
+
+        if (error_type === PASSWORD_ERRORS.RESET) {
```

Comment:
The functions getting a bit big. Maybe not enough to justify it's own file, but could this be it's own function/component?